odxtools/diaglayer.py

```diff
@@ -508,11 +575,15 @@ def get_tester_present_time(self) -> Optional[float]:
         (if no other request is sent to this ECU) in case of physically addressed requests."
         """
         com_param = self.get_communication_parameter("CP_TesterPresentTime")
-
         if com_param is None:
-            return 3.0  # default specified by the standard
+            return None
+
+        result = self._get_comparam_value(com_param)
+        if not result:
+            return None
+        assert isinstance(result, str)
 
-        return int(com_param.value) / 1e6
+        return int(float(result)/1e6)
```

Comment:
get_tester_present_time typehint says it should return float

Reply:
oops, fixed. (who specifies fractional seconds as tester present times anyway? ;))